### PR TITLE
Add `IBAnimatable 3.0 Migration Guide.md`.

### PR DESCRIPTION
**Pull request by JakeLin**
@tbaranes @lastMove Can you guys have a look at this Migration Guide? Please let me know if you find something is missing.

@huntit Can you have a look and see anything can be improved for the document?

### CODE REVIEW

File: Documentation/IBAnimatable 3.0 Migration Guide.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
+# IBAnimatable 3.0 Migration Guide
+
+IBAnimatable 3.0 is the latest major release of IBAnimatable, a library for designing and prototyping customized UI, interaction, navigation, transition and animation for App Store ready Apps in Interface Builder. As a major release, following Semantic Versioning conventions, 3.0 introduces API-breaking changes.
+
+This guide is provided in order to ease the transition of existing applications using IBAnimatable 2.x to the latest APIs, as well as explain the design and structure of new and updated functionality.
```
Comment (tbaranes) [inline Documentation/IBAnimatable 3.0 Migration Guide.md]: as explain -> as explaining?

File: Documentation/IBAnimatable 3.0 Migration Guide.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+- iOS 8.0+
+- Xcode 8.0+
+- Swift 3.0+
+
+For those of you that would like to use IBAnimatable with Swift 2.2 or 2.3, please use the latest tagged 2.x release which supports both Swift 2.2 and 2.3.
+
+## Benefits of upgrading
+
+- **Complete Swift 3 compatibility:** includes the full adoption of the new [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/).
+- **New enum system:** uses enums to provide nice programmatical APIs to replace stringy APIs.
```
Comment (tbaranes) [inline Documentation/IBAnimatable 3.0 Migration Guide.md]: stringy -> stringly ? Not sure 😆

File: Documentation/IBAnimatable 3.0 Migration Guide.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+*/
+var animationType: String? { get set }
+```  
+  
+After that we can use it for **Animation Type** in **Attributes inspector** like `SlideInLeft` or `SlideOutRight`. 
+
+And we can use it in code like: 
+
+```swift
+var view = AnimatableView() // Set up a           s view from Storyboard or frame.
```
Comment (tbaranes) [inline Documentation/IBAnimatable 3.0 Migration Guide.md]: There's a bunch of spaces in the comment

File: Documentation/IBAnimatable 3.0 Migration Guide.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+Since we have introduced new enum system, we have to change some properties like `animationType` and `maskType`. And we need to change them in the Storyboard when migrating version 3.
+
+You may see some warnings like below:
+> DemoApp.storyboard: warning: IB Designables: Ignoring user defined runtime attribute for key path "animationType" on instance of "IBAnimatable.AnimatableButton". Hit an exception when attempting to set its value: [<IBAnimatable.AnimatableButton 0x7ffc3a974800> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key animationType.
+
+Because we have change the properties name to support Interface Builder. Most of the cases, we can just add an underscore (`_`) to fix them.
+
+There are a couple of ways to fix them.
+
+0. Open the *.storyboard file with your favor editor e.g. Atom, VS Code or sublime. Replace the value as below:
```
Comment (tbaranes) [inline Documentation/IBAnimatable 3.0 Migration Guide.md]: You can also find them by doing a search in Xcode (even doing a replace all, but that's may be dangerous)

**Review by tbaranes — APPROVED** [review]
Looks good 👍

**Review by lastMove — APPROVED** [review]
All good.

**Comment by JakeLin** [comment]
Thanks @lastMove @tbaranes

**Pull request merged**